app/controllers/shared/api.php

```diff
@@ -49,44 +48,242 @@
     return $label;
 };
 
-$databaseListener = function (string $event, Document $document, Stats $usage) {
-    $multiplier = 1;
+$databaseListener = function (string $event, array $args, Document $project, Usage $queueForUsage, Database $dbForProject) {
+    $value = 1;
```

Comment:
Let's also make sure we have proper error handling inside this block so that if this fails, we log the issue and note break the API